.github/workflows/check-user.yml

```diff
@@ -0,0 +1,50 @@
+name: Check whether the workflow owner can use ARC runners
+
+on:
+  workflow_call:
+    inputs:
+      user_name:
+        required: true
+        type: string
+        description: The name of the workflow owner.
+      curr_branch:
+        required: true
+        type: string
+        description: Current branch.
+
+    outputs:
+      workflow-type:
+        description: Type of runners to use
+        value: ${{ jobs.check-user.outputs.workflow-type }}
+
+jobs:
+  check-user:
+    runs-on: linux.4xlarge
+    outputs:
+      workflow-type: ${{ steps.set-condition.outputs.workflow-type }}
+    steps:
+      - name: Check if user is in arc_users.yml
+        id: set-condition
+        run: |
```

Comment:
Might be better to put this run logic inside a python script. 
Way more versatile and easier to manage that way

Reply:
I thought about this. So far logic is pretty simple. If the solution becomes more cumbersome, I'll migrate it to a Python script.